flow/project.py

```diff
@@ -2054,13 +2043,103 @@ def _get_cached_status(self):
         except KeyError:
             return {}
 
-    def _get_operations_status(self, jobs, cached_status):
-        """Return a dict with information about job-operations for this aggregate.
+    def _generate_selected_aggregate_groups(
+        self,
+        selected_aggregates=None,
+        selected_groups=None,
+        tqdm_kwargs=None,
+    ):
+        """Yield selected aggregates and groups.
 
         Parameters
         ----------
-        jobs : :class:`~signac.contrib.job.Job` or aggregate of jobs
-            The signac job or aggregate.
+        selected_aggregates : iterable of aggregates
+            Aggregates to select.
+        selected_groups : set of :class:`~.FlowGroup`
+            Groups to select.
+        tqdm_kwargs : dict or None
+            A dict of keyword arguments to the tqdm progress bar used for
+            iterating over aggregates. If None, no progress bar will be
+            shown. (Default value = None)
+
+        Yields
+        ------
+        aggregate_id : str
+            Selected aggregate id.
+        aggregate : tuple of :class:`~signac.contrib.job.Job`
+            Selected aggregate.
+        group : :class:`~.FlowGroup`
+            Selected group.
+
+        """
+
+        def aggregate_progress_wrapper(aggregates):
+            """Show progress bar if keyword arguments are provided."""
+            if tqdm_kwargs is not None:
+                return tqdm(
+                    aggregates,
+                    total=len(aggregates),
+                    **tqdm_kwargs,
+                )
+            else:
+                return aggregates
+
+        if (
+            selected_groups is not None
+            and len(selected_groups) >= 0
+            and len(self._group_to_aggregate_store.inverse) > 1
+        ):
+            # Use only aggregate stores for the selected groups.
+            aggregate_stores_of_selected_groups = {
+                self._group_to_aggregate_store[group] for group in selected_groups
+            }
+            aggregate_stores = {
+                aggregate_store: self._group_to_aggregate_store.inverse[aggregate_store]
+                for aggregate_store in aggregate_stores_of_selected_groups
+            }
```

Comment:
```suggestion
            aggregate_stores = {
                aggregate_store: self._group_to_aggregate_store.inverse[aggregate_store]
                for group, aggregate_store in self._group_to_aggregate_store.items() 
                if group in selected_groups
            }
```

Can the code be replaced with this one?

Reply:
Also, `self._group_to_aggregate_store.inverse[aggregate_store]` would map to all the groups (even if they're not selected).

This means that we'll end up accepting all the groups if one of them is selected.

Reply:
@kidrahahjo I believe that your single comprehension is equivalent. I chose to do this in two comprehensions instead of one because I felt that one comprehension was a bit hard to read. If you think your suggestion is more readable, we can change it. I don't think there are any performance implications for either choice.

For your second question about mapping to all groups: yes, that is true. I see you noticed the comment on matching groups below, it was my intention to filter those groups (in order) in that later step. I don't think filtering at this point would make it any more efficient. We still perform the filtering before we iterate over the selected groups.

Reply:
Nah, I'm fine with it. I'll resolve this conversation.